backends/tfhe-cuda-backend/cuda/src/device.cu

```diff
@@ -52,184 +50,132 @@ void *cuda_malloc_async(uint64_t size, cuda_stream_t *stream) {
   return ptr;
 }
 
-/// Checks that allocation is valid
-/// 0: valid
-/// -1: invalid, not enough memory in device
-/// -2: invalid, gpu index doesn't exist
-int cuda_check_valid_malloc(uint64_t size, uint32_t gpu_index) {
-
-  if (gpu_index >= cuda_get_number_of_gpus()) {
-    // error code: invalid gpu_index
-    return -2;
-  }
-  cudaSetDevice(gpu_index);
+/// Check that allocation is valid
+void cuda_check_valid_malloc(uint64_t size, uint32_t gpu_index) {
+  check_cuda_error(cudaSetDevice(gpu_index));
   size_t total_mem, free_mem;
-  cudaMemGetInfo(&free_mem, &total_mem);
+  check_cuda_error(cudaMemGetInfo(&free_mem, &total_mem));
   if (size > free_mem) {
-    // error code: not enough memory
-    return -1;
+    printf("Cuda error: not enough memory on device. "
+            "Available: %d vs Requested: %d", free_mem, total_mem);
```

Comment:
%d is not the right specififer for size_t it seems https://stackoverflow.com/a/2125854/18088947

what sort of warning/lint level is enabled for the compiler on the cuda code ? Not saying we need to turn it on right now, but could be interesting to add more lints/warning levels as time goes on :)

Reply:
I fixed several other compilation warnings in this PR, but I need to make sure we don't let them pass in our PRs in the future.

Reply:
Here's the issue for this: https://github.com/zama-ai/tfhe-rs-internal/issues/434